### ISSUE TITLE
QZ not reading cadence from Costco proform tdf

### ISSUE THREAD

**Issue by MFR100** [cagnulein/qdomyos-zwift#2136]
[debug-Sat_Feb_17_10_26_54_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14319309/debug-Sat_Feb_17_10_26_54_2024.log)
**Describe the bug**
I think is not a big but a configuration issue.
I have a Costco Profirm TDF bike, a Samsung phone android 10 where QZ is installed, and either a pc or an android tablet where I run My Whoosh. Everything was working fine, but 2 weeks ago a failed system update on the phone left me with the only option to do a factory reset, so I lost all my apps and settings  
I reinstalled QZ and tried my best to get the settings, but I have 2 problems  
1. QZ connects ok to the bike, it sets the resistance to the start up value I have set in the app, but it does not detect any cadence or any other movement from the bike.
2. I am having problems to part My whoosh to the phone  The name of yhe.phone is BIKE but my whose times out
I attached the log file  
**To Reproduce**
1. Start pedaling so the bike starts
2. open QZ on the phone and see how the tiles update the resistance from 1 to 7
3. Listen and feel the bike resistance going up
4. keep pedaling
5. no speed or cadence or power are updated in QZ ui

**Expected behavior**
QZ detects my cadence, speed, and pass it to MyWhosh app on the pc

**Smartphone (please complete the following information):**
 - Device: Samsung galaxy S10
 - Android 10 

**Append a debug log**

 
[debug-Sat_Feb_17_10_26_54_2024.log](https://github.com/cagnulein/qdomyos-zwift/files/14319375/debug-Sat_Feb_17_10_26_54_2024.log)

  
  
**Additional context**

**Comment by cagnulein** [comment]
there is a main issue on the setting, under accessories, cadence sensor
must be set to disabled and press ok and restart qz
let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 17 feb 2024 alle ore 16:52 MFR100 ***@***.***>
ha scritto:

**Comment by cagnulein** [comment]
it could be a cache issue on mywhoosh. did you try to fully uninstall it?
in the log i saw that qz is sending bike metrics over bluetooth just fine

**Comment by MFR100** [comment]
Weird, I did it on the Android tablet with no results, so I did it on the Windows laptop with the same results, time out and no connection. I even tried to manually pair the phone to the laptop using the bluetooth setings and it pairs properly but for some reason MyWhoosh does not detect the device.
________________________________

**Comment by cagnulein** [comment]
let's understand if it's a phone issue or mywhoosh one. can you try the zwift trial version to check that's working there?

**Comment by MFR100** [comment]
OK, my Zwift trial expired some time ago, they also discontinued the free Wahoo RGT, so that is why the only free app I got now is MyWhoosh.
I tried login to Zwift from my tablet and it seems that it connecting recognizing the bike as Wahoo Kickr.
But I cannot try to  train because it asks me to subscribe.

________________________________

**Comment by cagnulein** [comment]
ok so in zwift do you see only the wahoo kickr? no "bike" device found? so it seems that the phone is not broadcasting bluetooth somehow. on mywhoosh don't you see the wahoo kickr?

**Comment by MFR100** [comment]
You are right in Zwift I only see the wahoo device, in MyWoosh I don't see any device

Miguel Rodriguez

________________________________

**Comment by MFR100** [comment]
Forgot to mention that in the same tablet I have Zwift and MyWoosh and only Zwift sees something. On the Windows laptop I only have MyWhoosh, I don't think there is a Zwift app for Windows, right?

________________________________

**Comment by cagnulein** [comment]
Yes there is a zwift windows app
While qz is open, are you able to open nrf connect app from another device
and send me a screenshot of how the phone is showing there?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 18 feb 2024 alle 00:59 MFR100 ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
Also on my whoosh you should see the Wahoo virtual device as well

![image](https://github.com/cagnulein/qdomyos-zwift/assets/48216/fd58351b-c490-4c0c-aa60-54f33d5aee22)

**Comment by MFR100** [comment]
Sorry, what is NRF?

Miguel Rodriguez

________________________________

**Comment by cagnulein** [comment]
Nrf connect is a free app on the store

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 18 feb 2024 alle 14:21 MFR100 ***@***.***> ha
scritto:

**Comment by MFR100** [comment]
Here you go

Miguel Rodriguez

________________________________

**Comment by cagnulein** [comment]
i can't see anything, you have to upload from GitHub directly

**Comment by cagnulein** [comment]
![image](https://github.com/cagnulein/qdomyos-zwift/assets/48216/71a871f4-6652-441b-a7df-79ba77af2a4b)

**Comment by cagnulein** [comment]
on nrfconnect can you click on connect on the bike device?

**Comment by MFR100** [comment]
Hi Roberto.  something weird happened. This morning, using the "BIE" phone I went to settings and did a reset on all connections, rebooted the phone, and then I tried 1.Starting a few pedals on the actual bike, 2.Start QZ on the "BIKE" phone, verified it was connecting and detecting cadence. 3. After reinstalling MyWooosh I rebooted the tablet. 4. Started MyWhoosh on the tablet. 5. started pairing from MYWhoosh to device and it paired without issues. 6. I quit MyWhoosh on the tablet, 7. Started MyWhoosh on the Windows laptop. 8. Paired frm MyWhoosh on windows to the devide and it paired without issues. 
Maybe it just needed to rest overnight? 
Anyways it seems to be fixed, I will keep using it next week and will let you know if there are any other surprises. 
Thanks a lot.
